### PR TITLE
feat(frontend/backend/shared): read/unread messages tm-501

### PR DESCRIPTION
**Pull request by YourTeaChan**
read-unread-messages

### CODE REVIEW

File: apps/frontend/src/modules/chats/libs/constants/one-unread-message.constant.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const ONE_UNREAD_MESSAGE = 1;
```
Comment (what1s1ove) [inline apps/frontend/src/modules/chats/libs/constants/one-unread-message.constant.ts:1]: specify a name. it is not clear what is should do

File: apps/frontend/src/libs/hooks/use-window-has-focus/use-window-has-focus.hook.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "~/libs/hooks/hooks.js";
+
+const useWindowHasFocus = (): boolean => {
```
Comment (what1s1ove) [inline apps/frontend/src/libs/hooks/use-window-has-focus/use-window-has-focus.hook.tsx:3]:
```suggestion
```
you do not need to handler realtime for two windows

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/chats/libs/components/chat/chat.tsx [hunk 3]
```diff
@@ -107,30 +117,32 @@ const Chat: React.FC<Properties> = ({
...
 						/>
 					</Link>
 					<span className={styles["full-name"]}>
-						{receiver.firstName} {receiver.lastName}
+						{interlocutor.firstName} {interlocutor.lastName}
 					</span>
 				</div>
 			</div>
 			<ul className={styles["chat-container"]}>
+				<span ref={lastMessageReference} />
```
Comment (what1s1ove) [inline apps/frontend/src/pages/chats/libs/components/chat/chat.tsx:136]: only `li` element can be inside `ul`

https://caninclude.glitch.me/

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/modules/chats/slices/chats.slice.ts [hunk 4]
```diff
@@ -74,34 +75,74 @@ const { actions, name, reducer } = createSlice({
...
+					return chat.id === newMessage.chatId
+						? {
+								...chat,
+								lastMessage: newMessage,
+								unreadMessageCount:
+									chat.interlocutor.id === newMessage.senderUser.id
+										? Number(chat.unreadMessageCount) + NEW_MESSAGE_COUNT
+										: chat.unreadMessageCount,
+							}
+						: chat;
```
Comment (Oleksandra-Masiuk) [inline apps/frontend/src/modules/chats/slices/chats.slice.ts:102]: what about writing like this?
```suggestion
					if(chat.id !== newMessage.chatId){
					return chat;
					}
					return {
								...chat,
								lastMessage: newMessage,
								unreadMessageCount:
									chat.interlocutor.id === newMessage.senderUser.id
										? Number(chat.unreadMessageCount) + NEW_MESSAGE_COUNT
										: chat.unreadMessageCount,
							};
```

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**